### PR TITLE
Simplify calculation of weights to run in ES-MDA

### PR DESCRIPTION
**Pull request by dafeda**
- [ ] PR title captures the intent of the changes, and is fitting for release notes.
- [ ] Added appropriate release note label
- [ ] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [ ] Make sure tests pass locally (after every commit!)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

File: src/ert/run_models/multiple_data_assimilation.py [hunk 1]
```diff
@@ -191,7 +189,7 @@ def run_experiment(
 
         self.setPhase(iteration_count + 1, "Experiment completed.")
 
-        return prior_context
+        return posterior_context
```
Comment (oyvindeide) [inline src/ert/run_models/multiple_data_assimilation.py:192]: Is this related to the other changes? Perhaps split it to another commit?

**Review by oyvindeide — APPROVED** [review]
LGTM! Just a minor comment

File: src/ert/run_models/multiple_data_assimilation.py (hunk 1 above)
Comment (dafeda) [reply src/ert/run_models/multiple_data_assimilation.py]: Not related no, I've removed it.
